### ISSUE TITLE
Build fails - 

### ISSUE THREAD

**Issue by petzlux** [Geoportail-Luxembourg/geoportailv3#269]
After a rebase from master, the build fails with:

````
Requirement already satisfied (use --upgrade to upgrade): WebOb==1.4 in ./.build/venv/lib/python2.7/site-packages (from -r CONST_versions.txt (line 74))
Requirement already satisfied (use --upgrade to upgrade): zope.component==4.2.1 in ./.build/venv/lib/python2.7/site-packages (from -r CONST_versions.txt (line 75))
Requirement already satisfied (use --upgrade to upgrade): zope.deprecation==4.1.2 in ./.build/venv/lib/python2.7/site-packages (from -r CONST_versions.txt (line 76))
Requirement already satisfied (use --upgrade to upgrade): zope.event==4.0.3 in ./.build/venv/lib/python2.7/site-packages (from -r CONST_versions.txt (line 77))
Requirement already satisfied (use --upgrade to upgrade): zope.interface==4.1.2 in ./.build/venv/lib/python2.7/site-packages (from -r CONST_versions.txt (line 78))
Requirement already satisfied (use --upgrade to upgrade): zope.sqlalchemy==0.7.5 in ./.build/venv/lib/python2.7/site-packages (from -r CONST_versions.txt (line 79))
Obtaining pyramid-closure from git+https://github.com/camptocamp/pyramid_closure#egg=pyramid_closure (from -r CONST_requirements.txt (line 5))
  Updating ./.build/venv/src/pyramid-closure clone
Obtaining pyramid from git+https://github.com/Pylons/pyramid@620bdefbff81792fd047f273a234264ce3d0a764#egg=pyramid (from -r CONST_requirements.txt (line 6))
  Updating ./.build/venv/src/pyramid clone (to 620bdefbff81792fd047f273a234264ce3d0a764)
  Could not find a tag or branch '620bdefbff81792fd047f273a234264ce3d0a764', assuming commit.
^@^@fatal: unable to connect to github.com:
github.com[0: 192.30.252.129]: errno=Connection timed out

Clone of 'git://github.com/Pylons/pylons_sphinx_theme.git' into submodule path 'docs/_themes' failed
  Complete output from command /bin/git submodule update --init --recursive -q:

  ----------------------------------------
  Command "/bin/git submodule update --init --recursive -q" failed with error code 1 in /home/wepa/geoportailv3/.build/venv/src/pyramid
make: *** [.build/requirements.timestamp] Error 1
````

The submodules fails to update, because it is specified by git protocol, not https protocol . Needs to be fixed 

Restant c2c 01/03/2015 : 1j

**Comment by elemoine** [comment]
@sbrunner is going to look into it.

**Comment by petzlux** [comment]
Build still fails !

This is an upstream issue with the pyramid package. It defines it submodule like this https://github.com/Pylons/pyramid/blob/master/.gitmodules#L3 

Because its git not https protocol, the git submodule update fails.

I managed to manually update the relevant submodule by changing https://github.com/Pylons/pyramid/blob/master/.gitmodules#L3 to https instead of git

**Comment by sbrunner** [comment]
Je viens de créer une branche sur un clone de pyramid ou je n'utilise pas le protocole git.

donc si vous remplace dans `CONST_requirements.txt`
`-e git+https://github.com/Pylons/pyramid@1e02bbfc0df09259bf207112acf019c8dba44a90#egg=pyramid`
par 
`-e git+https://github.com/sbrunner/pyramid@c2cgeoportal#egg=pyramid`
es-ce que cela fonctionne ?

**Comment by elemoine** [comment]
Ce bug est-il toujours d'actualité ?

**Comment by jaykayone** [comment]
non, c'est OK

**Comment by jaykayone** [comment]
currently using a workaround with a fork from sbrunner.
better solution is necessary

**Comment by petzlux** [comment]
workaround here, ideally raise issue in pyramid repo and change to https protocl 
https://github.com/Geoportail-Luxembourg/geoportailv3/blob/info_bar/CONST_requirements.txt#L6

**Comment by sbrunner** [comment]
I reported the issue but it's not going to be in pyramid ...
https://github.com/Pylons/pyramid/issues/1596

**Comment by elemoine** [comment]
I am not surprised. There must be a better way :-)

**Comment by elemoine** [comment]
I'm adding estimates for this bug. Camptocamp to find a solution…

**Comment by sbrunner** [comment]
Why is it reopend ?

**Comment by jaykayone** [comment]
good question, I don't know.
we'll wait for @petzlux to be back

**Comment by elemoine** [comment]
The problem still exists for the Luxembourg team. See https://github.com/Geoportail-Luxembourg/geoportailv3/pull/403#issuecomment-92364394 from @rmichaelis.

**Comment by rmichaelis** [comment]
After a rebase, I get this error again.
Has something reintroduced the error? 
Or was the error still there ?

>Cloning https://github.com/camptocamp/pyramid_closure to ./.build/venv/src/pyramid-closure
Obtaining pyramid from git+https://github.com/Pylons/pyramid@1e02bbfc0df09259bf207112acf019c8dba44a90#egg=pyramid (from -r CONST_requirements.txt (line 6))
  Cloning https://github.com/Pylons/pyramid (to 1e02bbfc0df09259bf207112acf019c8dba44a90) to ./.build/venv/src/pyramid
>Could not find a tag or branch '1e02bbfc0df09259bf207112acf019c8dba44a90', assuming commit.
fatal: unable to connect to github.com:
github.com[0: 192.30.252.130]: errno=Connection timed out
>Clone of 'git://github.com/Pylons/pylons_sphinx_theme.git' into submodule path 'docs/_themes' failed
>Complete output from command git submodule update --init --recursive -q:

>  ----------------------------------------
>  Command "git submodule update --init --recursive -q" failed with error code 1 in /home/mire/geoportailv3/.build/venv/src/pyramid
>make: *** [.build/requirements.timestamp] Error 1

**Comment by elemoine** [comment]
Strange. Could you please copy&paste the content of your `CONST_requirements.txt` file?

**Comment by rmichaelis** [comment]
here it is :
--index-url http://pypi.camptocamp.net/pypi
--find-links http://pypi.camptocamp.net/internal-pypi/index/c2cgeoportal
--find-links http://pypi.camptocamp.net/internal-pypi/index/c2cgeoportal-win
-r CONST_versions.txt
-e git+https://github.com/camptocamp/pyramid_closure#egg=pyramid_closure
-e git+https://github.com/Pylons/pyramid@1e02bbfc0df09259bf207112acf019c8dba44a90#egg=pyramid
-e .

**Comment by rmichaelis** [comment]
It's not the same as the one in the master

**Comment by elemoine** [comment]
Yes, indeed, I was going to say the same. It sounds like you rebased onto an old master. Maybe you did `git rebase master` and your local `master` branch is not up-to-date. Try `git fetch origin` and then `git rebase origin/master` instead.

**Comment by rmichaelis** [comment]
Ok, it sounds like a git problem.
I was explaining my problem to Patrick. 
I showed him the bad file in my branch. 
Then I executed in this order 
 `git fetch origin` 
 `git co master` 
 `git merge --ff-only origin/master `

The result of the git merge never showed that CONST_requirements.txt has changed.
But now CONST_requirements.txt  is the good one, both in the master and in my branch...

I really not understand, but the build works now.

Thanks for the support
